gitbook/network-ftp.md

```diff
@@ -2,189 +2,186 @@
+Every function of the [Ftp]({{book.api}}/Ftp.html) class wraps an FTP command, and returns a standard FTP response. An FTP response contains a status code (similar to HTTP status codes but not identical), and a message that informs the user of what happened. FTP responses are encapsulated in the [Ftp::Response]({{book.api}}/Ftp::Response.html) class.
 
-```
-#include <SFML/Network.hpp>
+```ruby
+ftp = SF::Ftp.new
 
-sf::Ftp ftp;
 ...
-sf::Ftp::Response response = ftp.login(); // just an example, could be any function
 
-std::cout << "Response status: " << response.getStatus() << std::endl;
-std::cout << "Response message: " << response.getMessage() << std::endl;
+response = ftp.login("username", "password") # just an example, could be any function
+
+puts "Response status: " + response.status.to_s
+puts "Response message: " + response.message
 ```
 
-The status code can be used to check whether the command was successful or failed: Codes lower than 400 represent success, all others represent errors. You can use the `isOk()` function as a shortcut to test a status code for success. 
+The status code can be used to check whether the command was successful or failed: Codes lower than 400 represent success, all others represent errors. You can use the `ok?` function as a shortcut to test a status code for success.
 
-```
-sf::Ftp::Response response = ftp.login();
-if (response.isOk())
-{
-    // success!
-}
+```ruby
+response = ftp.login("username", "password")
+if response.ok?
+    # success!
 else
-{
-    // error...
-}
+    # error...
+end
 ```
 
-If you don't care about the details of the response, you can check for success with even less code: 
+If you don't care about the details of the response, you can check for success with even less code:
 
-```
-if (ftp.login().isOk())
-{
-    // success!
-}
+```ruby
+if ftp.login("username", "password").ok?
+    # success!
 else
-{
-    // error...
-}
+    # error...
+end
 ```
 
-For readability, these checks won't be performed in the following examples in this tutorial. Don't forget to perform them in your code! 
+For readability, these checks won't be performed in the following examples in this tutorial. Don't forget to perform them in your code!
 
-Now that you understand how the class works, let's have a look at what it can do. 
+Now that you understand how the class works, let's have a look at what it can do.
 
 ## Connecting to the FTP server
 
-The first thing to do is connect to an FTP server. 
+The first thing to do is connect to an FTP server.
 
-```
-sf::Ftp ftp;
-ftp.connect("ftp.myserver.org");
+```ruby
+ftp = SF::Ftp.new
+ftp.connect("ftp.myserver.org")
```

Comment:
I'm not planning to polish Network module in the nearest future. I only generated it, but barely touched it.
If you want to continue with this, you'd have to look at what the defaults are
http://www.sfml-dev.org/documentation/2.3.1/classsf_1_1Ftp.php
and supply them manually.